flink-connector-mysql-cdc/src/main/java/com/ververica/cdc/connectors/mysql/source/assigners/ChunkSplitter.java

```diff
@@ -239,50 +265,31 @@ private MySqlSnapshotSplit createSnapshotSplit(
     // ------------------------------------------------------------------------------------------
 
     /** Checks whether split column is evenly distributed across its range. */
-    private static boolean isSplitColumnEvenlyDistributed(
-            JdbcConnection jdbc,
-            TableId tableId,
-            Column splitColumn,
-            Object min,
-            Object max,
-            int chunkSize)
-            throws SQLException {
+    private static boolean isEvenlySplitColumn(Column splitColumn) {
         DataType flinkType = MySqlTypeUtils.fromDbzColumn(splitColumn);
         LogicalTypeRoot typeRoot = flinkType.getLogicalType().getTypeRoot();
 
         // currently, we only support the optimization that split column with type BIGINT, INT,
         // DECIMAL
-        if (!(typeRoot == LogicalTypeRoot.BIGINT
+        return typeRoot == LogicalTypeRoot.BIGINT
                 || typeRoot == LogicalTypeRoot.INTEGER
-                || typeRoot == LogicalTypeRoot.DECIMAL)) {
-            return false;
-        }
+                || typeRoot == LogicalTypeRoot.DECIMAL;
+    }
 
-        if (ObjectUtils.minus(max, min).compareTo(BigDecimal.valueOf(chunkSize)) <= 0) {
-            return true;
+    /** Returns the dynamic chunk size of evenly distributed table. */
+    private static int getDynamicChunkSize(
+            int chunkSize, double distributionFactor, double evenlyDistributionFactor) {
+        int dynamicChunkSize = chunkSize;
+        if (distributionFactor <= evenlyDistributionFactor) {
+            dynamicChunkSize = Math.max((int) (distributionFactor * chunkSize), dynamicChunkSize);
```

Comment:
It seems there is no tests cover this. Would be better add a test for the large dist factor case.

Reply:
OK. Thanks, I just think an IT case is hard to verify the splitted chunks.